broker/tasks/pipelines.py

```diff
@@ -136,11 +137,13 @@ def queue_all_cdn_dedicated_waf_provision_tasks_for_operation(
         .then(letsencrypt.answer_challenges, operation_id, **correlation)
         .then(letsencrypt.retrieve_certificate, operation_id, **correlation)
         .then(iam.upload_server_certificate, operation_id, **correlation)
-        .then(cloudfront.create_distribution, operation_id, **correlation)
+        .then(waf.create_web_acl, operation_id, **correlation)
+        .then(cloudfront.create_distribution_with_tags, operation_id, **correlation)
         .then(cloudfront.wait_for_distribution, operation_id, **correlation)
         .then(route53.create_ALIAS_records, operation_id, **correlation)
         .then(route53.wait_for_changes, operation_id, **correlation)
-        .then(waf.create_web_acl, operation_id, **correlation)
+        .then(route53.create_health_checks, operation_id, **correlation)
+        .then(shield.associate_health_checks, operation_id, **correlation)
```

Comment:
This might not work because the Shield protection is added automatically by AWS and might not be ready by the time this step is reached, but it seemed worth trying because other approaches are much more complex